File: src/main/java/fi/quanfoxes/types/Normal.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Normal extends Number {
    private static final int BYTES = 4;

    public Normal() {
        super(NumberType.INT32, 32, "num");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Normal.cs
public class Normal : Number
{
	private const int BYTES = 4;

	public Normal() : base(NumberType.INT32, 32, false, "normal") { }

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
